python/cudf/cudf/core/column/string.py

```diff
@@ -5938,15 +5930,15 @@ def view(self, dtype) -> "cudf.core.column.ColumnBase":
         str_end_byte_offset = self.base_children[0].element_indexing(
             self.offset + self.size
         )
-        char_dtype_size = self.base_children[1].dtype.itemsize
+        char_dtype_size = cudf.api.types.dtype("int8").itemsize
 
         n_bytes_to_view = (
             str_end_byte_offset - str_byte_offset
         ) * char_dtype_size
 
         to_view = column.build_column(
-            self.base_children[1].data,
-            dtype=self.base_children[1].dtype,
+            self.base_data,  # self.base_children[1].data,
+            dtype=cudf.api.types.dtype("int8"),  # self.base_children[1].dtype,
```

Comment:
Why not `self.base_data.dtype`?

Reply:
`self.base_data` is cudf.core.buffer. Buffer type. It does not have a dtype.